src/components/organisms/NewResourceWizard/NewResourceWizard.tsx

```diff
@@ -75,20 +75,18 @@ const NewResourceWizard = () => {
 
   const kindsByApiVersion = useMemo(
     () =>
-      ResourceKindHandlers.reduce((result, resource) => {
-        const foundResourceKindHandler = getResourceKindHandler(resource.kind);
-
-        if (foundResourceKindHandler) {
-          if (result[resource.clusterApiVersion] && result[resource.clusterApiVersion]) {
-            result[resource.clusterApiVersion].push(foundResourceKindHandler);
-          } else {
-            result[resource.clusterApiVersion] = [foundResourceKindHandler];
-          }
+      ResourceKindHandlers.reduce((result, resourcekindHandler) => {
+        if (result[resourcekindHandler.clusterApiVersion]) {
+          result[resourcekindHandler.clusterApiVersion].push(resourcekindHandler);
+        } else {
+          result[resourcekindHandler.clusterApiVersion] = [resourcekindHandler];
         }
 
         return result;
       }, {} as Record<string, ResourceKindHandler[]>),
-    []
+    // depend on resourceMap since newly loaded resources could have contained CRDs that resulted in dynamically
+    // created kindHandlers
```

Comment:
@devcatalin see comment in code - it would probably be better to have the kindhandlers be part of the app state so we can depend on those directly - is this what you are planning to solve the section blueprints?